src/libData/BlockData/BlockHeader/VCBlockHeader.cpp

```diff
@@ -99,28 +97,27 @@ const vector<pair<PubKey, Peer>>& VCBlockHeader::GetFaultyLeaders() const {
 };
 
 bool VCBlockHeader::operator==(const VCBlockHeader& header) const {
-  return (
-      (m_VieWChangeDSEpochNo == header.m_VieWChangeDSEpochNo) &&
-      (m_VieWChangeEpochNo == header.m_VieWChangeEpochNo) &&
-      (m_ViewChangeState == header.m_ViewChangeState) &&
-      (m_CandidateLeaderNetworkInfo == header.m_CandidateLeaderNetworkInfo) &&
-      (m_CandidateLeaderPubKey == header.m_CandidateLeaderPubKey) &&
-      (m_VCCounter == header.m_VCCounter) &&
-      (m_FaultyLeaders == header.m_FaultyLeaders));
+  return BlockHeaderBase::operator==(header) &&
+         (std::tie(m_VieWChangeDSEpochNo, m_VieWChangeEpochNo,
+                   m_ViewChangeState, m_CandidateLeaderNetworkInfo,
+                   m_CandidateLeaderPubKey, m_VCCounter, m_FaultyLeaders) ==
+          std::tie(header.m_VieWChangeDSEpochNo, header.m_VieWChangeEpochNo,
+                   header.m_ViewChangeState,
+                   header.m_CandidateLeaderNetworkInfo,
+                   header.m_CandidateLeaderPubKey, header.m_VCCounter,
+                   header.m_FaultyLeaders));
 }
 
 bool VCBlockHeader::operator<(const VCBlockHeader& header) const {
   // To compare, first they must be of identical epochno and state
-  return (m_VieWChangeDSEpochNo == header.m_VieWChangeDSEpochNo) &&
-         (m_VieWChangeEpochNo == header.m_VieWChangeEpochNo) &&
-         (m_ViewChangeState == header.m_ViewChangeState) &&
+  return BlockHeaderBase::operator==(header) &&
+         (std::tie(m_VieWChangeDSEpochNo, m_VieWChangeEpochNo,
```

Comment:
Should version be included?

Reply:
version must be equal and which is included in the BlockHeaderBase